### DISCUSSION TITLE
installing with Apache

### DISCUSSION THREAD

**Discussion by freedomtrain** [searx/searx#852] [Installation issue: custom]
I have Searx running in virtualenv
I have apache running 
How do I demonize Searx with apache
I have read the documentation but I m confused 
Do I need to follow uwsgi instructions and then follow the with apache section  ?

**Comment by kvch** [comment]
No, you should skip the uwsgi section and read the section apache.

**Comment by dalf** [comment]
uwsgi demonizes searx.
apache (or nginx) is the frontend which allow to setup HTTPS connection (uwsgi is not a frontend).

So you need to follow : 
* Basic installation
* Configuration
* Check
* uwsgi
* Web server / with apache

**Comment by kvch** [comment]
@fredhampton My bad. @dalf is completely right. I somehow thought you were referring to nginx. :)

**Comment by freedomtrain** [comment]
@dalf 

I followed the steps but seem to be missing something concerning apache
Where is the place in apache2.conf to add 
<Location /usr/local/searx>
	Options FollowSymLinks Indexes
    SetHandler uwsgi-handler
    uWSGISocket /run/uwsgi/app/searx/socket
</Location>

Do I need to create a searx.conf file ?

Thanks for your help

**Comment by freedomtrain** [comment]
I followed the steps but seem to be missing something concerning apache
Where is the place in apache2.conf to add
<Location /usr/local/searx>
Options FollowSymLinks Indexes
SetHandler uwsgi-handler
uWSGISocket /run/uwsgi/app/searx/socket
Do I need to create a searx.conf file ?

Thanks for your help

**Comment by freedomtrain** [comment]
I added below to apache2.conf and could not reach searx on my severs localhost:8888
The uwsgi searx.ini is working.... I can see the entires in the log file....
Does it make any difference that I m using plesk on debian 8 ?
<Location /usr/local/searx>
	Options FollowSymLinks Indexes
    SetHandler uwsgi-handler
    uWSGISocket /run/uwsgi/app/searx/socket
</Location>

**Comment by dalf** [comment]
Not sure but I guess that ```<Location /usr/local/searx>``` should be replaced by ```<Location />``` or ```<Location /searx>``` .

**Comment by freedomtrain** [comment]
My searx files are here /use/local/searx   
If this is the case should location be this  <Location /> or <Location /searx> or <Location /usr/local/searx>

**Comment by dalf** [comment]
I can't double check right now. But uwsgi is used so the searx install directory is not important except for the static files. So /searx

**Comment by freedomtrain** [comment]
also receive this error message when running python webapp.py

Traceback (most recent call last):
  File "webapp.py", line 48, in <module>
    from flask import (
  File "/usr/local/lib/python2.7/dist-packages/flask/__init__.py", line 19, in <module>
    from jinja2 import Markup, escape
  File "/usr/local/lib/python2.7/dist-packages/jinja2/__init__.py", line 33, in <module>
    from jinja2.environment import Environment, Template
  File "/usr/local/lib/python2.7/dist-packages/jinja2/environment.py", line 15, in <module>
    from jinja2 import nodes
  File "/usr/local/lib/python2.7/dist-packages/jinja2/nodes.py", line 19, in <module>
    from jinja2.utils import Markup
  File "/usr/local/lib/python2.7/dist-packages/jinja2/utils.py", line 624, in <module>
    from markupsafe import Markup, escape, soft_unicode
ImportError: No module named markupsafe
